src/commands/deployGateway.ts

```diff
@@ -0,0 +1,213 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import { env, OpenDialogOptions, ProgressLocation, Uri, window, workspace } from "vscode";
+import { DialogResponses } from '../../extension.bundle';
+import { IGatewayToken } from '../azure/apim/contracts';
+import { GatewayTreeItem } from "../explorer/GatewayTreeItem";
+import { ext } from "../extensionVariables";
+import { localize } from "../localize";
+import { cpUtils } from "../utils/cpUtils";
+import { openUrl } from '../utils/openUrl';
+import { requestTest } from '../utils/requestUtil';
+
+// tslint:disable-next-line: export-name
+export async function copyRunDockerCommand(node?: GatewayTreeItem): Promise<void> {
+    if (!node) {
+        node = <GatewayTreeItem>await ext.tree.showTreeItemPicker(GatewayTreeItem.contextValue);
+    }
+
+    ext.outputChannel.show();
+    ext.outputChannel.appendLine(localize("deployGateway", "Generating command for running gateway in docker..."));
+
+    window.withProgress(
+        {
+            location: ProgressLocation.Notification,
+            title: localize("DeployGateway", "Generate command for running in Docker."),
+            cancellable: true
+        },
+        async () => {
+            // tslint:disable: no-non-null-assertion
+            ext.outputChannel.appendLine(localize("deployGateway", "Checking components installed..."));
```

Comment:
you dont need these checks anymore as its a copy command and we should not care if they have docker.